src/Bridge/Symfony/Bundle/Resources/config/swagger.xml

```diff
@@ -38,14 +38,41 @@
             <tag name="serializer.normalizer" priority="17" />
         </service>
 
+        <service id="api_platform.openapi.normalizer.documentation" class="ApiPlatform\Core\OpenAPI\Serializer\DocumentationNormalizer" public="false">
+            <argument type="service" id="api_platform.metadata.resource.metadata_factory" />
+            <argument type="service" id="api_platform.metadata.property.name_collection_factory" />
+            <argument type="service" id="api_platform.metadata.property.metadata_factory" />
+            <argument type="service" id="api_platform.resource_class_resolver" />
+            <argument type="service" id="api_platform.operation_method_resolver" />
+            <argument type="service" id="api_platform.operation_path_resolver" />
+            <argument>null</argument>
+            <argument type="service" id="api_platform.filter_locator" />
+            <argument type="service" id="api_platform.name_converter" on-invalid="ignore" />
+            <argument>%api_platform.oauth.enabled%</argument>
+            <argument>%api_platform.oauth.type%</argument>
+            <argument>%api_platform.oauth.flow%</argument>
+            <argument>%api_platform.oauth.tokenUrl%</argument>
+            <argument>%api_platform.oauth.authorizationUrl%</argument>
+            <argument>%api_platform.oauth.scopes%</argument>
+            <argument>%api_platform.swagger.api_keys%</argument>
+            <argument type="service" id="api_platform.subresource_operation_factory" />
+            <argument>%api_platform.collection.pagination.enabled%</argument>
+            <argument>%api_platform.collection.pagination.page_parameter_name%</argument>
+            <argument>%api_platform.collection.pagination.client_items_per_page%</argument>
+            <argument>%api_platform.collection.pagination.items_per_page_parameter_name%</argument>
+            <argument>%api_platform.collection.pagination.client_enabled%</argument>
+            <argument>%api_platform.collection.pagination.enabled_parameter_name%</argument>
+            <tag name="serializer.normalizer" priority="18" />
```

Comment:
What do you think about allowing to choose the OpenAPI version through a query parameter? As done for API Gateway here: https://github.com/api-platform/core/blob/master/src/Documentation/Action/DocumentationAction.php#L66

Then it will be easy to test, and convenient for end users (they will be able to use OpenAPI v3 by default when supported, but still fallback to v2 for old tools).

For Swagger UI, we should to the same thing IMO (defaulting to v3, but allowing to switch to v2 by setting a query parameter).

WDYT?

Reply:
looks like a good idea, I'll try to update soon